samza-core/src/main/java/org/apache/samza/system/inmemory/InMemorySystemFactory.java

```diff
@@ -31,20 +35,28 @@
  * Initial draft of in-memory {@link SystemFactory}. It is test only and not meant for production use right now.
  */
 public class InMemorySystemFactory implements SystemFactory {
-  private static final InMemoryManager MEMORY_MANAGER = new InMemoryManager();
+  private static Map<Integer, InMemoryManager> jobToManager = new HashMap<>();
 
   @Override
   public SystemConsumer getConsumer(String systemName, Config config, MetricsRegistry registry) {
-    return new InMemorySystemConsumer(MEMORY_MANAGER);
+    jobToManager.putIfAbsent(parseJobId(config), new InMemoryManager());
+    return new InMemorySystemConsumer(jobToManager.get(parseJobId(config)));
   }
 
   @Override
   public SystemProducer getProducer(String systemName, Config config, MetricsRegistry registry) {
-    return new InMemorySystemProducer(systemName, MEMORY_MANAGER);
+    jobToManager.putIfAbsent(parseJobId(config), new InMemoryManager());
+    return new InMemorySystemProducer(systemName, jobToManager.get(parseJobId(config)));
```

Comment:
Just curious.

Would using `jobId` here work for high level API job? 

Besides functionality it will be better to not use JobId here. It's  looks odd to use JobId  in a factory for producer/consumer(In memory producer and consumer could be used outside of a samza job and doesn't have to be a part of a samza job).

Reply:
Replaced `jobId` with `test.id`. The guarantees of this param is, if configured it will provide an independent instance of InMemorySystem to run, if not the application uses a default static InMemorySystem which is shared across all the classes